src/main/java/de/tum/in/www1/artemis/web/rest/NotificationResource.java

```diff
@@ -100,6 +105,10 @@ public ResponseEntity<Notification> createNotification(@RequestBody Notification
     @PreAuthorize("hasRole('USER')")
     public ResponseEntity<List<Notification>> getAllNotificationsForCurrentUserFilteredBySettings(@ApiParam Pageable pageable) {
         User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        var tutorialGroups = tutorialGroupService.findAllForNotifications(currentUser);
+        HashSet<Long> tutorialGroupIds = new HashSet<>();
+        tutorialGroups.forEach(tutorialGroup -> tutorialGroupIds.add(tutorialGroup.getId()));
```

Comment:
I think you can also put them in a stream and map them to their Ids and collect them to a set

Reply:
Good idea! Done in bf2dbc60b7c8745b90ebc393c71b31b17bc183e4